app/src/main/java/at/bitfire/icsdroid/ui/EditCalendarActivity.kt

```diff
@@ -96,60 +99,170 @@ class EditCalendarActivity: AppCompatActivity() {
-                .setEnabled(!dirty)
-                .setVisible(!dirty)
-
-        menu.findItem(R.id.cancel)
-                .setEnabled(dirty)
-                .setVisible(dirty)
-
-        // if local file, hide authentication fragment
-        val uri = model.subscriptionWithCredential.value?.subscription?.url
-        binding.credentials.visibility =
-            if (uri != null && HttpUtils.supportsAuthentication(uri))
-                View.VISIBLE
-            else
-                View.GONE
-
-        val titleOK = !subscriptionSettingsModel.title.value.isNullOrBlank()
-        val authOK = credentialsModel.run {
-            if (requiresAuth.value == true)
-                username.value != null && password.value != null
-            else
-                true
+    @Composable
+    private fun EditCalendarComposable() {
+        val url by subscriptionSettingsModel.url.observeAsState("")
+        val title by subscriptionSettingsModel.title.observeAsState("")
+        val color by subscriptionSettingsModel.color.observeAsState(0)
+        val ignoreAlerts by subscriptionSettingsModel.ignoreAlerts.observeAsState(false)
+        val defaultAlarmMinutes by subscriptionSettingsModel.defaultAlarmMinutes.observeAsState()
+        val defaultAllDayAlarmMinutes by subscriptionSettingsModel.defaultAllDayAlarmMinutes.observeAsState()
+        val inputValid by inputValid.observeAsState(false)
+        val modelsDirty by modelsDirty.observeAsState(false)
+        Scaffold(
+            topBar = { AppBarComposable(inputValid, modelsDirty) }
+        ) { paddingValues ->
+            Column(
+                Modifier
+                    .verticalScroll(rememberScrollState())
+                    .padding(paddingValues)
+                    .padding(16.dp)
+            ) {
+                SubscriptionSettingsComposable(
+                    url = url,
+                    title = title,
+                    titleChanged = { subscriptionSettingsModel.title.postValue(it) },
+                    color = color,
+                    colorIconClicked = { colorPickerContract.launch(color) },
+                    ignoreAlerts = ignoreAlerts,
+                    ignoreAlertsChanged = { subscriptionSettingsModel.ignoreAlerts.postValue(it) },
+                    defaultAlarmMinutes = defaultAlarmMinutes,
+                    defaultAlarmMinutesChanged = {
+                        subscriptionSettingsModel.defaultAlarmMinutes.postValue(
+                            it.toLongOrNull()
+                        )
+                    },
+                    defaultAllDayAlarmMinutes = defaultAllDayAlarmMinutes,
+                    defaultAllDayAlarmMinutesChanged = {
+                        subscriptionSettingsModel.defaultAllDayAlarmMinutes.postValue(
+                            it.toLongOrNull()
+                        )
+                    },
+                    // TODO: Complete with some valid state
+                    isCreating = false,
```

Comment:
For this, I'd create a new `LiveData` in `EditSubscriptionModel`, which holds when the `ViewModel` is saving, inside the `updateSubscription` function. Maybe inside a try-finally?

Reply:
Not sure, why we need this? DB updates happen almost instantaneously. I don't think the user will have time to change the collection again in that time and even if they did, why would that pose a problem?

Reply:
I just wanted it to be more informative on what's happening, so when it's updating the database, everything is disabled, notifying the user that the request is being performed. It's instant, of course, but maybe in some shitty phone it's not the case, I don't know.

Even though it's true that it's not the same for creation than updates, it's much quicker.

Whatever the case, we should remove the TODO 😅

Reply:
> Even though it's true that it's not the same for creation than updates, it's much quicker.

Not sure what you mean here. What is not the same and/or quicker? :)

> I just wanted it to be more informative on what's happening, so when it's updating the database, everything is disabled, notifying the user that the request is being performed. It's instant, of course, but maybe in some shitty phone it's not the case, I don't know.

It would disable everything for only a few ms max, so this might not even be visible feedback for the user. Quite sure its the same on slow devices. I got one for testing and even there db updates happen instantly.

For user feedback we already have [the toast](https://github.com/ArnyminerZ/icsx5/pull/46/files#diff-4d77a4cfab3683b0123ba975d0c00894bfc88c0ffaadd3280faca6df2e6ff7afR137) being shown. Furthermore after editing or creating a subscription the view immediately changes to the subscriptions overview. So in order to see the disabled textboxes the user has to re-open the `EditCalendarActivity` fast enough.

In case it really becomes an issue I am happy for us to add it, later on. Right now I think this feature creates more code and maintenance than its worth. :slightly_smiling_face: 

> Whatever the case, we should remove the TODO 😅

Will do :+1:

Reply:
> Whatever the case, we should remove the TODO 😅

Done :) 

If you feel alright about adding the `isCreating` mechanism later, if needed, and are happy with the PR otherwise, we can merge now :+1:

Reply:
I think it's not necessary. I'll merge